gradoop-store/gradoop-accumulo/src/main/java/org/gradoop/common/storage/impl/accumulo/iterator/tserver/GradoopVertexIterator.java

```diff
@@ -0,0 +1,161 @@
+/**
+ * Copyright © 2014 - 2018 Leipzig University (Database Research Group)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradoop.common.storage.impl.accumulo.iterator.tserver;
+
+import org.apache.accumulo.core.data.Key;
+import org.apache.accumulo.core.data.Range;
+import org.apache.accumulo.core.data.Value;
+import org.apache.accumulo.core.iterators.SortedKeyValueIterator;
+import org.apache.accumulo.core.util.Pair;
+import org.gradoop.common.model.api.entities.EPGMVertex;
+import org.gradoop.common.model.impl.id.GradoopId;
+import org.gradoop.common.model.impl.id.GradoopIdSet;
+import org.gradoop.common.model.impl.pojo.Vertex;
+import org.gradoop.common.model.impl.pojo.VertexFactory;
+import org.gradoop.common.model.impl.properties.PropertyValue;
+import org.gradoop.common.storage.impl.accumulo.constants.AccumuloTables;
+import org.gradoop.common.utils.KryoUtils;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.io.IOException;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Objects;
+
+/***
+ * accumulo gradoop vertex iterator
+ */
+public class GradoopVertexIterator extends BaseElementIterator<EPGMVertex> {
+
+  /**
+   * vertex factory
+   */
+  private final VertexFactory factory = new VertexFactory();
+
+  @Nonnull
+  @Override
+  public Vertex fromRow(@Nonnull Map.Entry<Key, Value> pair) throws IOException {
+    //map from serialize content
+    Vertex content = KryoUtils.loads(pair.getValue().get(), Vertex.class);
+    content.setId(GradoopId.fromString(pair.getKey().getRow().toString()));
+    //read from content
+    return content;
+  }
+
+  @Nonnull
+  @Override
+  public Pair<Key, Value> toRow(@Nonnull EPGMVertex record) throws IOException {
+    //write to content
+    return new Pair<>(new Key(record.getId().toString()),
+      new Value(KryoUtils.dumps(factory.initVertex(
+        record.getId(),
+        record.getLabel(),
+        record.getProperties(),
+        record.getGraphIds()))));
+  }
+
+  @Nonnull
+  @Override
+  public Iterator<EPGMVertex> doSeek(
```

Comment:
I really don't like this anonymous way of returning an iterator. Can we restructure this into a more clear and easier to maintain version?

Reply:
Good remark. Same for Edge and GraphHead